### PR TITLE
0.1.0

### PR DESCRIPTION
**Pull request by MoOx**
- es6 via babel
- eslint only with stricter rules
- tests are in `**/__tests__/*` to be close to original source (ignored for
publish via npmignore)

I tried to make things simple so we can iterate quickly

@davidtheclark lgty ?

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,10 +1,27 @@
...
+
+env:
+  es6: true
+  browser: true
+  node: true
+
+# 0: off, 1: warning, 2: error
+rules:
+  # semicolons are useless
+  semi: [2, "never"]
```
Comment (magsout) [inline .eslintrc]: For CSS ?

Comment (MoOx) [inline .eslintrc]: haha nooo. eslint lint js only man :)

Comment (magsout) [inline .eslintrc]: .... ahaha my bad... hard to read on my iPhone ;)

**Comment by MoOx** [comment]
appveyor doesn't like the symlink, not a big deal for now.

**Comment by davidtheclark** [comment]
Interesting -- I haven't yet written a Node module using ES6 (just browser stuff that needs compilation anyway). Seems very straightforward, though, and if you have enjoyed it then sounds good to me.

Location of tests is fine with me.

Re: ESLint --- two questions:
- Do we want to add *more* rules to enforce the things that JSCS was enforcing (e.g. spacing)? At some point we could do that to enforce the current code style.
- Are you really attached to no semicolons? That's the only part I don't like; and I'd worry that it would make the codebase seem weird to potential contributors.

**Comment by tunnckoCore** [comment]
> it would make the codebase seem weird to potential contributors.

it's not weird, see [feross/standard](https://github.com/feross/standard) weird is that spacing between function name and parentheses, but I can go forward and I start following and applying the standard for my repos.
It would be good you and that project also follow it.

**Comment by jeddy3** [comment]
It's great to see you two kicking this project off! I've looked through both the ES5 and ES6 code for this first rule and I've learnt a lot about JavaScript already.

Are you guys OK if we name this rule `declaration-block-trailing-semicolon` as the semicolon doesn't really belong to the value, but rather to the declaration block? I guess that's because each declaration can have multiple (comma separated) values. I know that's probably a bit pedantic, but I think it'd be nice if we established a good naming pattern right from the beginning. 

To help us I wrote [this gist](https://gist.github.com/jeddy3/6ca5fa8196fb8d1a90fb) near the end of our rules discussion. It lists as many of the rules we spoke about as I could remember, and I tried to group them based on the specs - note the likes of `String`, `Number`, `Function`, `Color`. We didn't discuss the groupings at the time though, so I'm not sure what you both think of them. Some of the names a little more verbose (although some are now shorter), but they all more closely match the specs and follow a nice pattern. Hopefully the groupings are OK with you both? 

The `trailing-semicolon` is the 2nd rule in [`declaration-block` group](https://gist.github.com/jeddy3/6ca5fa8196fb8d1a90fb#declaration-block-rules). I'm more than happy to do a PR with the name change to the ES6 code, if you're both OK with it. It would be a great opportunity for me to get started with the code base.

Also, I tried to build and test this pull request locally, but I got the same error as AppVeyor as I'm developing on windows. I think I fixed it locally (tweaks to the npm run-scripts commands), so I'm going to try to push up it up now. Hopefully I'll do it right!

**Comment by davidtheclark** [comment]
I am fine with the rename. I thought I was following the names you wrote down, but I guess I was looking at an earlier version or something.

**Comment by MoOx** [comment]
@davidtheclark 

> Do we want to add more rules to enforce the things that JSCS was enforcing (e.g. spacing)? At some point we could do that to enforce the current code style.

Not feeling the need. After using jscs for months, I feel it doesn't bring any value. On the other hand I found eslint default very nice to prevent errors and which add minimal rules for styles (eg: `thing="stuff"` is invalid and should be `thing = "stuff"`).
I just added some rules for opininated styles. Because it's cool to have some :)

> Are you really attached to no semicolons? That's the only part I don't like; and I'd worry that it would make the codebase seem weird to potential contributors.

They are just useless. Lot's of language don't use them. Even bootstrap don't use semicolon :trollface: 

@jeddy3 I will take a deeper look to the naming thing tomorrow before merging that.

**Comment by MoOx** [comment]
Again, thanks a lot @davidtheclark for this first rules.
And thanks everyone for the feedback.

**Comment by tunnckoCore** [comment]
@davidtheclark im looking the gist now, but `function-opening-parenthesis-after` and `function-opening-parenthesis-before` looks strange for me.

they mean that?
```css
a {
  width: calc( 2px * 2 );
}
```
this looks ugly and dont have logic.

it would be good to write examples for them.

**Comment by jeddy3** [comment]
@davidtheclark No worries at all :) The rules discussion got pretty messy towards the end, with lists of rules all over that thread! I tried to tidy it all up in my [final comment](https://github.com/stylelint/stylelint/issues/1#issuecomment-72226871) as I was getting quite confused.

Oh, and like MoOx said, thanks for kicking off the code for this first rule :)

@MoOx Please don't hesitate to give me a shout if you have a questions about the naming thing. Btw, there is some rationale in that [final comment](https://github.com/stylelint/stylelint/issues/1#issuecomment-72226871), especially about standardising how the `no-*` rules might behave. If you do have any questions, perhaps we should discuss them in the [rules thread](https://github.com/stylelint/stylelint/issues/1) to keep this PR focused on the code.

@tunnckoCore That gist is just a summary of all the rules from the [rules thread](https://github.com/stylelint/stylelint/issues/1).  MoOx came up with the original list and we all added to it as the discussion went on. Most of the rules were first attempts and not final, especially all the `*-after/before` stuff. I'll try to answer your `function-opening-parenthesis-after` question in the [rules thread](https://github.com/stylelint/stylelint/issues/1) a little later, as it's probably worth trying to keep all the detailed discussions about the rules in once place.

**Pull request merged**